atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,690 @@
+
+    def parse_system_info(self):
+        self._system_info = {'system': {}, 'calculation': {}}
+        particles_group = self._data_parser.get(self._path_group_particles_all)
+        positions = self._data_parser.get(self._path_value_positions_all)
+        n_frames = self._n_frames
+        if particles_group is None or positions is None or positions is None: # For now we require that positions are present in the H5MD file to store other particle attributes
+            self.logger.warning('No positions available in H5MD file. Other particle attributes will not be stored')
+            return self._system_info
+
+        def get_value(value, steps, path=''):
+            if value is None:
+                return value
+            if isinstance(value, h5py.Group):
+                path_value = f'{path}.value' if path else 'value'
+                value = self._data_parser.get(path_value)
+                path_step = f'{path}.step' if path else 'step'
+                attr_steps = self._data_parser.get(path_step)
+                if value is None or attr_steps is None:
+                    self.logger.warning('Missing values or steps in particle attributes. These attributes will not be stored.')
+                    return None
+                elif attr_steps.sort() != steps.sort() if attr_steps is not None else False:
+                    self.logger.warning('Distinct trajectory lengths of particle attributes not supported. These attributes will not be stored.')
+                    return None
+                else:
+                    return value
+            else:
+                return [value] * n_frames
+
+        # get the steps based on the positions
+        steps = self._data_parser.get(f'{self._path_group_positions_all}.step')
+        if steps is None:
+            self.logger.warning('No step information available in H5MD file. System information cannot be parsed.')
+            return self._system_info
+        self.trajectory_steps = steps
+
+        # get the rest of the particle quantities
+        values_dict = {'system': {}, 'calculation': {}}
+        times = self._data_parser.get(f'{self._path_group_positions_all}.time')
+        values_dict['system']['time'] = times
+        values_dict['calculation']['time'] = times
+        values_dict['system']['positions'] = positions
+        values_dict['system']['n_atoms'] = self._n_atoms
+        system_keys = {'labels': 'system', 'velocities': 'system', 'forces': 'calculation'}
+        for key, sec_key in system_keys.items():
+            path = f'{self._path_group_particles_all}.{self._nomad_to_particles_group_map[key]}'
+            value = self._data_parser.get(path)
+            values_dict[sec_key][key] = get_value(value, steps, path=path)
+
+        # get the box quantities
+        box = self._data_parser.get(f'{self._path_group_particles_all}.box')
+        if box is not None:
+            box_attributes = {'dimension': 'system', 'periodic': 'system'}
```

Comment:
will we have a case where the section key is not system? if not why use dict? is it the case that you want it to be consistent with l234

Reply:
I think probably not. Yes, I was just trying to be consistent and leave it general for the future, but I think now it's not necessary. I changed box_attributes to a list now.